### PR TITLE
Added offer soldering.

### LINKED ISSUE #9670: Wiring framework: create offer solder connection
Soldering currently has the option to use `put()` strategy or `inject()` strategy. It would be nice if we also supported `offer()` where a failed offer causes the data to be discarded.

### PR DESCRIPTION
**Pull request by cody-littley**
Closes #9670 

Must merge after https://github.com/hashgraph/hedera-services/pull/9664

### CODE REVIEW

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/model/StandardWiringModel.java [hunk 1]
```diff
@@ -150,11 +150,11 @@ public void registerEdge(
             @NonNull final String originVertex,
             @NonNull final String destinationVertex,
             @NonNull final String label,
-            final boolean injection) {
+            final boolean nonBLocking) {
```
Comment (alittley) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/model/StandardWiringModel.java:153]: nOnBLoCkiNg

**Review by alittley — DISMISSED** [review]

**Review by edward-swirldslabs — DISMISSED** [review]

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/model/StandardWiringModel.java (hunk 1 above)
Comment (cody-littley) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/model/StandardWiringModel.java:153]: fixed

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/WiringModel.java [hunk 2]
```diff
@@ -151,11 +151,11 @@ public final TaskSchedulerMetricsBuilder metricsBuilder() {
...
      * @param label             the label of the edge
-     * @param injection         true if this edge is an injection edge, false otherwise
+     * @param nonBLocking       true if this edge is an injection edge or an offer edge, false otherwise
      */
     public abstract void registerEdge(
             @NonNull String originVertex,
             @NonNull String destinationVertex,
             @NonNull String label,
-            final boolean injection);
+            final boolean nonBLocking);
```
[the remark this answers is not in the document]
Comment (cody-littley) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/WiringModel.java:160]: fixed

[the remark this answers is not in the document]
Comment (cody-littley) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/WiringModel.java:160]: Good suggestion, done.

**Review by alittley — APPROVED** [review]

**Pull request merged**